File: api/models/Salle.js
const mongoose = require("mongoose");

const SalleSchema = new mongoose.Schema({
    nom: { type: String, required: true },
    type: { type: String, required: true, enum: ["noir", "blanc"] },
    machines: [{ type: mongoose.Schema.Types.ObjectId, ref: "Machine" }]
});

module.exports = mongoose.model("Salle", SalleSchema);

File: frontend_flutter/lib/models/salle.dart
import 'machine.dart';

class Salle {
  String id;
  String nom;
  String type;
  List<Machine>? machines;

  Salle({
    required this.id,
    required this.nom,
    required this.type,
    this.machines,
  });

  factory Salle.fromJson(Map<String, dynamic> json) {
    return Salle(
      id: json['_id'],
      nom: json['nom'],
      type: json['type'],
      machines: json['machines'] != null
          ? (json['machines'] as List).map((m) => Machine.fromJson(m)).toList()
          : [],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      '_id': id,
      'nom': nom,
      'type': type,
      'machines': machines?.map((m) => m.toJson()).toList(),
    };
  }
}
